### ISSUE TITLE
CA1845 fixer doesn't recognize range syntax

### ISSUE THREAD

**Issue by Neme12** [dotnet/roslyn-analyzers#6905]
### Analyzer

**Diagnostic ID**: [CA1845](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1845): `Use span-based 'string.Concat'

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: SDK 8.0.100-preview.7.23376.3

**Version**: 5.0.3 (Latest)

### Describe the bug
The following code generates [CA1845](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1845): `Use span-based 'string.Concat`:
```c#
void Foo(string str, int position)
{
    var foo = str[0..position] + str.Substring(position + 1);
}
```
and invoking the code fix changes it to the following:
```c#
void Foo(string str, int position)
{
    var foo = string.Concat(str[0..position], str.AsSpan(position + 1));
}
```
Note that the fixer correctly replaced `Substring` with `AsSpan`, but didn't recognize the range syntax that could be changed to use `AsSpan` as well. I believe it should produce the following code:
```c#
void Foo(string str, int position)
{
    var foo = string.Concat(str.AsSpan(0..position), str.AsSpan(position + 1));
}
```

**Comment by Tinister** [comment]
I've also noticed that CA1845 doesn't trigger when you use the range syntax as your Substring.

That is to say that `a + b.Substring(1)` triggers CA1845, but `a + b[1..]` does not trigger CA1845, despite the latter just being a fancier way of invoking the same Substring method.

Not sure if that applies to this issue or if it's worth creating as a separate issue.
